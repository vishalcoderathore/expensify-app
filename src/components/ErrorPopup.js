import React from 'react';

class ErrorPopup extends React.Component{
    render(){
        return(
            <div>
                <p>Please enter Description and Amount</p>
            </div>
        )
    }
}

export default ErrorPopup;