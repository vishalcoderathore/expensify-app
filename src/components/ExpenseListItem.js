import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({description, amount, createdAt, note, dispatch, id}) => (
    <div>
        
        <Link to={`/edit/${id}`}><h1>Description: {description}</h1></Link>
        <p>Amount: {amount} ,Created on: {createdAt}</p>
        <p>Expense Note: {note}</p>
       
    </div>
        
);
export default ExpenseListItem;

