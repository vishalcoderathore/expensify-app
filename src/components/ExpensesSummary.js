import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';
import expenses_total from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/visibleExpenses';

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
    const formattedExpenseTotal = numeral(expenseTotal).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseCount === 1 ? 'expense' : 'expenses'} totalling <span>{formattedExpenseTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: expenses_total(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);