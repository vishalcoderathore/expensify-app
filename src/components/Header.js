import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';


export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link to="/dashboard" className="header__title">
                    <img src="/images/expensify-logo.png" />
                </Link>
                <button onClick={startLogout} className="header__button">Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
};

export default connect(undefined, mapDispatchToProps)(Header); 