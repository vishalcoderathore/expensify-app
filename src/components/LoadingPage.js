import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/lg.dual-gear-loading-icon.gif"/>
    </div>
);

export default LoadingPage;