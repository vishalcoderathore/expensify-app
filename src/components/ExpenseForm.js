import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import ErrorPopup from './ErrorPopup';

const now = moment();

class ExpenseForm extends React.Component{
    
    constructor(props){
        super(props);
        this.state={
            description     :    this.props.expense ? this.props.expense.description : '',
            amount          :    this.props.expense ? (this.props.expense.amount).toString() : '',
            note            :    this.props.expense ? this.props.expense.note : '',
            createdAt       :    this.props.expense ? moment(this.props.expense.createdAt) : moment(),
            calenderFocused :    false,
            error           :    false  
        }
    }

    onDescChange = (e) => {
        const updatedDesc = e.target.value;
        this.setState(() => {
           return {
               description : updatedDesc
           }
       });
    };
    onAmountChange = (e) => {
        const updatedAmt = e.target.value;
        if(!updatedAmt || updatedAmt.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(() => {
                return {
                    amount : updatedAmt
                }
            });
        }
    };
    onNoteChange = (e) => {
        const updatedNote = e.target.value;
        this.setState(() => {
           return {
               note : updatedNote
           }
       });
    };
    onDateChange = (createdAt) => {
        if(createdAt){
            this.setState(() => ({createdAt : createdAt}));
        }
    }
    onFocusChange = ({focused}) => {
        this.setState(() => ({calenderFocused : focused}));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(() => ({error : true}));
        }
        else{
            this.setState(() => ({error : false}));
            this.props.onSubmit({
                description : this.state.description,
                amount : parseFloat(this.state.amount, 10),
                createdAt : this.state.createdAt.valueOf(),
                note : this.state.note 
            });
        }
    };


    render(){
        return (
            <div>
                {this.state.error && <ErrorPopup />}
                <form onSubmit={this.onSubmit}>
                
                    <input type="text" id="desc" placeholder="Description" value={this.state.description} onChange={this.onDescChange} autoFocus/>
                    
                    <input type="text" placeholder="Amount" value={this.state.amount} onChange = {this.onAmountChange}/>
                    
                    <SingleDatePicker
                        date = {this.state.createdAt} 
                        onDateChange = {this.onDateChange}
                        focused = {this.state.calenderFocused}
                        onFocusChange = {this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange = {() => false}
                    />
                    
                    <textarea placeholder="Add a note for your Expense" cols="30" rows="10" value={this.state.note} onChange={this.onNoteChange}></textarea>
                
                    <button>{this.props.expense ? 'Update Expense' : 'Add Expense'}</button>
                
                </form>
            </div>
        )
    }
} 

export default ExpenseForm;