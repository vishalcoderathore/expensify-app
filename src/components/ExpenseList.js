import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/visibleExpenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? <p>No expenses Added</p> : 
                props.expenses.map((expense) => {
                    return <ExpenseListItem  key={expense.id} {...expense}/>
                })
        }
    </div>
);

const mapStateToProps = (state) => {
    return{
        expenses: selectExpenses(state.expenses, state.filters)
    };
};


export default connect(mapStateToProps)(ExpenseList);
