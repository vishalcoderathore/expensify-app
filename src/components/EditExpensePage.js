import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startEditExpense } from '../actions/expenses';
import { startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove Expense</button>
            </div>
        );
    }
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

// Removes the usage of :
// this.props.dispatch(editExpense(this.props.expense.id, expense));
// and
// this.props.dispatch(startRemoveExpense({id : this.props.expense.id}));
const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);