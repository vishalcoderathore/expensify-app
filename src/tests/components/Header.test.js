import React from 'react';
import { Header } from '../../components/Header';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

let wrapper;
// beforeEach(() => {
//     wrapper = shallow(<Header startLogout={() => { }} />);
// });

test('should render header correctly', () => {
    wrapper = shallow(<Header startLogout={() => { }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();

});