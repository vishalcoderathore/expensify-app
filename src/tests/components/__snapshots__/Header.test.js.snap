// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <img
          height="50"
          src="/images/expensify-logo.png"
          width="380"
        />
      </Link>
      <button
        className="header__button"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
